services/provider/server/server.go

```diff
@@ -418,10 +418,17 @@ func (s *OCSProviderServer) GetDesiredClientState(ctx context.Context, req *pb.G
 
 // OffboardConsumer RPC call to delete the StorageConsumer CR
 func (s *OCSProviderServer) OffboardConsumer(ctx context.Context, req *pb.OffboardConsumerRequest) (*pb.OffboardConsumerResponse, error) {
-	err := s.consumerManager.Delete(ctx, req.StorageConsumerUUID)
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get storageConsumer resource with the provided UUID. %v", err)
+	}
+	storageConsumer.Status.Client = &ocsv1alpha1.ClientStatus{}
+	err = s.consumerManager.client.Update(ctx, storageConsumer)
```

Comment:
```suggestion
	err = s.client.Status.Update(ctx, storageConsumer)
```
and make the error inline